### PR TITLE
attestation-service: Replace anyhow error crate with thiserror crate

### PR DESCRIPTION
**Pull request by kartikjoshi21**

### CODE REVIEW

File: attestation-service/attestation-service/src/bin/restful-as.rs [hunk 1]
```diff
@@ -79,27 +99,31 @@ async fn main() -> Result<()> {
...
     let server = match (cli.https_prikey, cli.https_pubkey_cert) {
         (Some(prikey), Some(pubkey_cert)) => {
-            let mut builder = SslAcceptor::mozilla_modern(SslMethod::tls())?;
+            let mut builder = SslAcceptor::mozilla_modern(SslMethod::tls())
+                                .map_err(|e| RestfulError::OpenSSL(e.to_string()))?;
 
             let prikey = tokio::fs::read(prikey)
                 .await
-                .context("read HTTPS private key")?;
+                .map_err(|e| RestfulError::ReadKey(e.to_string()))?;
```
Comment (mkulke) [inline attestation-service/attestation-service/src/bin/restful-as.rs:107]: I don't think we need to explicity map the errors, shouldn't a entry in the enum be sufficient?

```rust
#[error("io error")]
IO(#[from] std::io::Error) 
```

similarly for the other errors, we can look at the error types that the fn return and wrap them like this.

**Comment by Xynnn007** [comment]
Hi @kartikjoshi21 Thanks for the replacement. I love the detailed error types and I'd like to take a review once you make this ready. : )

**Review by mkulke — COMMENTED** [review]
I don't think we need all the `map_err()` we should wrap the original errors in the error enum instead

File: attestation-service/attestation-service/src/rvps/mod.rs [hunk 2]
```diff
@@ -70,15 +84,15 @@ impl RvpsConfig {
                             warn!("No RVPS address provided and will launch a built-in rvps");
                             Ok(Box::new(builtin::Rvps::new(&self.store_type)?) as Box<dyn RvpsApi + Send + Sync>)
                         } else {
-                            Err(anyhow!("either feature `rvps-grpc` or `rvps-builtin` should be enabled."))
+                            return RvpsError::FeatureNotEnabled;
```
[the remark this answers is not in the document]
Comment (kartikjoshi21) [reply attestation-service/attestation-service/src/rvps/mod.rs:87]: Yess sure Magnus, iam going to go through all the errors again and will return original error types.

**Review by mkulke — COMMENTED** [review]
There are a lot of error types added in this PR, I don't think we should duplicate all errors of underlying crates. 

If we have a thiserror enum:

```rust
#[derive(Error, Debug)]
pub enum MyError {
    #[error("Some library error: {0}")]
    SomeLibrary(#[from] SomeLibrary::Error),
    ...
}
```

then we can use `?` for implicit conversion, thiserror will take care of this, no need to duplicate all the error cases of `SomeLibrary::Error` in our library:

```rust
fn my_fn() -> Result<(), MyError> {
    SomeLibrary::do_something()?;
    Ok(())
}
```

**Comment by kartikjoshi21** [comment]
Distributing this to smaller  PR

**Pull request closed without merging**